test/load/chatLoadTest.js

```diff
@@ -0,0 +1,78 @@
+const WebSocket = require('ws');
+const fetch = require('node-fetch');
+
+var connectionCount = 0;
+const targetConnectionCount = 5000;
+
+const messages = [
+  'I am a test message',
+  'this is fake',
+  'i write emoji 😀',
+  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
+  'Sed pulvinar proin gravida hendrerit. Mauris in aliquam sem fringilla ut morbi tincidunt augue. In cursus turpis massa tincidunt dui.',
+  'Feugiat in ante metus dictum at tempor commodo ullamcorper. Nunc aliquet bibendum enim facilisis gravida neque convallis a. Vitae tortor condimentum lacinia quis vel eros donec ac odio.',
+  'Here is _some_ **markdown**!',
+];
+
+var availableMessages = messages.slice();
+
+
+async function registerChat() {
+  const options = {
+      method: 'POST',
+      headers: {
+          'Content-Type': 'application/json'
+      }
+  }
+
+  try {
+      const response = await fetch('http://localhost:8080/api/chat/register', options);
+      const result = await response.json();
+      return result;
+  } catch(e) {
+      console.error(e);
+  }
+}
+
+async function sendMessage() {
```

Comment:
From a review POV, it's not clear what is happening when. Is this function only running once? Is it calling itself (with a timeout)? What is the `send` function doing inside of it?

Reply:
I renamed and added comments it to be clearer.  I could totally spend a bunch of time building a far cleaner script, but this just needs to do one specific thing to assist in testing, so it was thrown together for that because the load testing framework wouldn't allow me to do what I needed to do.  Hopefully we can get rid of it if Artillery's update comes out some day.